pkg/distributor/forwarding/forwarding.go

```diff
@@ -339,7 +344,8 @@ func (f *forwarder) splitToIngestedAndForwardedTimeseries(tsSliceIn []mimirpb.Pr
 			if filteredSamples > 0 {
 				err = errSamplesTooOld
 				if !ingest {
-					f.discardedSamplesTooOld.WithLabelValues(user).Add(float64(filteredSamples))
+					f.activeGroups.UpdateActiveGroupTimestamp(user, group, time.Now())
```

Comment:
I would say we need to update the group as soon as we see it in the request, not only if it errors.